File: ios/Classes/SDK/AdContentInfoMethodHandler.swift
//
//  AdContentInfoHandler.swift
//  clever_ads_solutions
//
//  Copyright © 2025 CleverAdsSolutions LTD, CAS.AI. All rights reserved.
//

import CleverAdsSolutions
import Flutter

private let channelName = "cleveradssolutions/ad_content_info"

class AdContentInfoMethodHandler: MappedMethodHandler<AdContentInfo> {
    init(with registrar: FlutterPluginRegistrar) {
        super.init(with: registrar, on: channelName)
    }

    override func onMethodCall(_ instance: AdContentInfo, _ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        let value: Any?
        switch call.method {
        case "getFormat":
            value = instance.format.value
        case "getSourceName":
            value = instance.sourceName
        case "getSourceId":
            value = instance.sourceID
        case "getSourceUnitId":
            value = instance.sourceUnitID
        case "getCreativeId":
            value = instance.creativeID
        case "getRevenue":
            value = instance.revenue
        case "getRevenuePrecision":
            value = instance.revenuePrecision
        case "getImpressionDepth":
            value = instance.impressionDepth
        case "getRevenueTotal":
            value = instance.revenueTotal
        default:
            super.onMethodCall(instance, call, result)
            return
        }

        result(value)
    }
}

File: android/src/main/kotlin/com/cleveradssolutions/plugin/flutter/sdk/AdContentInfoMethodHandler.kt
package com.cleveradssolutions.plugin.flutter.sdk

import com.cleveradssolutions.plugin.flutter.bridge.base.MappedMethodHandler
import com.cleveradssolutions.sdk.AdContentInfo
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

private const val CHANNEL_NAME = "cleveradssolutions/ad_content_info"

class AdContentInfoMethodHandler(binding: FlutterPluginBinding) :
    MappedMethodHandler<AdContentInfo>(binding, CHANNEL_NAME) {

    override fun onMethodCall(
        instance: AdContentInfo,
        call: MethodCall,
        result: MethodChannel.Result
    ) {
        val value = with(instance) {
            when (call.method) {
                "getFormat" -> format.value
                "getSourceName" -> sourceName
                "getSourceId" -> sourceId
                "getSourceUnitId" -> sourceUnitId
                "getCreativeId" -> creativeId
                "getRevenue" -> revenue
                "getRevenuePrecision" -> revenuePrecision
                "getImpressionDepth" -> impressionDepth
                "getRevenueTotal" -> revenueTotal
                else -> return super.onMethodCall(instance, call, result)
            }
        }
        result.success(value)
    }

}